src/ORM/Marshaller.php

```diff
@@ -133,7 +138,7 @@ public function one(array $data, array $options = [])
             $columnType = $schema->columnType($key);
             if (isset($propertyMap[$key])) {
                 $assoc = $propertyMap[$key]['association'];
-                $value = $this->_marshalAssociation($assoc, $value, $propertyMap[$key]);
+                $value = $this->_marshalAssociation($assoc, $value, $propertyMap[$key] + $marshallOptions);
```

Comment:
I think HasMany associations will load the existing records much like belongsToMany, and have the same issues with 'sync' scenarios.